test/celeritas/optical/Cerenkov.test.cc

```diff
@@ -0,0 +1,200 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/Cerenkov.test.cc
+//---------------------------------------------------------------------------//
+#include <vector>
+
+#include "corecel/cont/Range.hh"
+#include "corecel/data/CollectionBuilder.hh"
+#include "corecel/math/Algorithms.hh"
+#include "celeritas/Constants.hh"
+#include "celeritas/Units.hh"
+#include "celeritas/grid/GenericGridData.hh"
+#include "celeritas/optical/CerenkovDndxCalculator.hh"
+#include "celeritas/optical/CerenkovGenerator.hh"
+#include "celeritas/optical/CerenkovParams.hh"
+
+#include "DiagnosticRngEngine.hh"
+#include "celeritas_test.hh"
+
+namespace celeritas
+{
+namespace test
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Tabulated refractive index in water as a function of photon wavelength [μm].
+ *
+ * M. Daimon and A. Masumura. Measurement of the refractive index of distilled
+ * water from the near-infrared region to the ultraviolet region, Appl. Opt.
+ * 46, 3811-3820 (2007) via refractiveindex.info
+ *
+ * See G4OpticalMaterialProperties.hh.
+ */
+std::vector<double> const& get_wavelength()
+{
+    static std::vector<double> const wavelength = {
```

Comment:
This still ends up copying to the heap at runtime: instead, use `double const wavelength[]` and return a `Span`. (Assuming the data will eventually end up in the main source, that is: if it's really only for the unit test, then don't worry about it.)

Reply:
Ah thanks. It's possible this will end up in the main source... I'll update it in any case.